### PR TITLE
#32: AetherRepositoryTest

### PR DESCRIPTION
**Pull request by Nimmaster**
issue #32

Makes `AetherRepository#upload` work.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#33](https://github.com/artipie/maven-adapter/issues/33) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/4be7421c-89a2-4959-8cf0-a966a914b06d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by Nimmaster** [comment]
@g4s8 pls escalate it

**Comment by g4s8** [comment]
@0crat assign @paulodamaso

File: src/main/java/com/artipie/maven/aether/AetherRepository.java [hunk 1]
```diff
@@ -159,15 +156,24 @@ private DeployResult deploy(final String path, final InputStream content) throws
                 try (var file = Files.newOutputStream(staging.unwrap())) {
                     content.transferTo(file);
                 }
+                final var artifact = new DefaultArtifact(
+                    new FileCoordinates(path).coords()
+                )
+                    .setFile(
```
Comment (paulodamaso) [inline src/main/java/com/artipie/maven/aether/AetherRepository.java]: @Nimmaster Please correct this indentation here, it is a bit weird.

File: src/main/java/com/artipie/maven/aether/AetherRepository.java [hunk 2]
```diff
@@ -159,15 +156,24 @@ private DeployResult deploy(final String path, final InputStream content) throws
...
-                            ).setFile(
-                                staging.unwrap().toFile()
+                        .addArtifact(artifact)
+                        .setRepository(
+                            new RemoteRepository.Builder(
+                                "asto",
+                                "default",
+                                "asto://storage/maven-adapter"
                             )
+                                .build()
```
Comment (paulodamaso) [inline src/main/java/com/artipie/maven/aether/AetherRepository.java]: @Nimmaster Here also, indent it properly

File: src/main/java/com/artipie/maven/aether/AstoTransporter.java [hunk 3]
```diff
@@ -95,6 +87,24 @@ public void implClose() {
...
+    /**
+     * Checks if task resource exists.
+     * @param task TransportTask
+     * @throws ResourceNotFoundException if task resource does not exist
+     */
+    private void exists(final TransportTask task) {
+        final Key key = new TaskKey(task).key();
+        if (!this.asto.exists(key)) {
+            throw new ResourceNotFoundException(
+                String.format(
```
Comment (paulodamaso) [inline src/main/java/com/artipie/maven/aether/AstoTransporter.java]: @Nimmaster Use `cactoos` `Formatted` insted `String.format`

File: src/test/java/com/artipie/maven/aether/AetherRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            new AutoCloseablePath.Parent(this.temp.resolve("staging"))
+        );
+    }
+
+    @Test
+    public void shouldUpload() throws Exception {
+        final var path = "org/example/artifact/1.0/artifact-1.0.jar";
+        final var bytes = new byte[0];
+        final var artifact = this.repository.upload(path, new ByteArrayInputStream(bytes));
+        MatcherAssert.assertThat(
```
Comment (paulodamaso) [inline src/test/java/com/artipie/maven/aether/AetherRepositoryTest.java]: @Nimmaster This method is testing three different things, please split it into three different methods.

File: src/main/java/com/artipie/maven/aether/AstoTransporter.java (hunk 3 above)
Comment (Nimmaster) [reply src/main/java/com/artipie/maven/aether/AstoTransporter.java]: I just extracted out the piece of code to a new method. When I added it in the first place, @g4s8 decided that Cactoos is not mandatory to use.

**Comment by 0crat** [comment]
Job [#33](https://github.com/artipie/maven-adapter/issues/33) is already in scope

**Comment by 0crat** [comment]
> @0crat assign @paulodamaso  ([here](https://github.com/artipie/maven-adapter/issues/33#issuecomment-587008403))

@g4s8 This pull request [#33](https://github.com/artipie/maven-adapter/issues/33) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/b4cfe80e-a62a-42c5-99f7-a631ca6d7268); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
GitHub issue is still open, won't close

**Comment by 0crat** [comment]
Job [#33](https://github.com/artipie/maven-adapter/issues/33) is not in the agenda of @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), can't retrieve data and time of add

File: src/main/java/com/artipie/maven/aether/AstoTransporter.java (hunk 3 above)
Comment (g4s8) [reply src/main/java/com/artipie/maven/aether/AstoTransporter.java]: @paulodamaso @Nimmaster I don't see the difference between `new Formatted().asString()` and `String.format()` here. I'd use `Text` when lazy evaluation is needed.

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by paulodamaso** [comment]
@g4s8 Good to merge

**Comment by 0crat** [comment]
Job was finished in 2 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**